### PR TITLE
Fix dump when using comptime variable val.${field.name}

### LINKED ISSUE #16489: cgen: Alias problem
### Describe the bug

The problem is that I can't get the type of the Alias. When I try dump(val.$(field.name)) I have a cgen error

### Expected Behavior

0 or ''

### Current Behavior

```bash

╰─ v run alias.v ─╯
==================
/tmp/v_1000/alias.8312079028855434033.tmp.c:12329: error: cannot convert 'struct string' to 'struct main__NotAlias'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Reproduction Steps

```v
type ID = string
type GG = int
type JJ = int

struct Alias {
	id ID
	gg GG
	equal GG
	jj JJ
}

struct NotAlias {
	id string
	ij int
	equal int
	jj int
}

fn encode_struct<U>(val U){
	println('-------------------------------')
	$for field in U.fields {
        dump(field)
        dump(val.$(field.name))
    }
}
alias := Alias{}
not_alias := NotAlias{}

encode_struct(alias)
encode_struct(not_alias)
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

latest 

### Environment details (OS name and version, etc.)

ubuntu

### LINKED ISSUE #16631: cgen: `$for`
### Describe the bug

val.$(field.name) problem

### Expected Behavior

all work

### Current Behavior

not work

### Reproduction Steps

```v
module main

struct Foo{
	a string 
	b ?string 
}

fn sei_oq_la[T](val T){
	$for field in T.fields {
		$if field.typ is string  {
			dump(val.$(field.name)) // work "c"
		}
		$if field.typ is ?string {
			dump(val.$(field.name)) // work "d"
		}
		$if field.typ is string || field.typ is ?string {
			dump(val.$(field.name)) // break // builder error: // /tmp/v_1000/v_bug.3977126037910564797.tmp.c:12157: error: field not found: data
		}
	}
}

fn main(){
	foo := Foo{"c","d"}
	sei_oq_la(foo)
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.2 1ba1f99

### Environment details (OS name and version, etc.)

Ubuntu

### PR DESCRIPTION
**Pull request by felipensp**
Fix #16631
Fix #16489

```V
module main

struct Foo{
	a string 
	b ?string
	c int
	d ?int
	e struct {
		i ?int
	}
}

fn sei_oq_la[T](val T){
	$for field in T.fields { dump(val.$(field.name)) }
}

fn main(){
	foo := Foo{"c","d", 1, 0, struct {100}}
	sei_oq_la(foo)
}
```

### CODE REVIEW

File: vlib/v/checker/tests/comptime_dump_fields_var_test.vv [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	a string
+	b ?string
+	c int
+	d ?int
+	e struct {
+		i ?int
+	}
+}
+
+fn sei_oq_la[T](val T) {
```
Comment (spytheman) [inline vlib/v/checker/tests/comptime_dump_fields_var_test.vv:11]: what does `sei_oq_la` mean?

Comment (spytheman) [reply vlib/v/checker/tests/comptime_dump_fields_var_test.vv:11]: oh, it is directly from the issue 🤔

Comment (felipensp) [reply vlib/v/checker/tests/comptime_dump_fields_var_test.vv:11]: It's a portuguese phrase for "whatever". I was just copying from the original test case's issue.

**Comment by felipensp** [comment]
> Maybe can, also, fix: #16489 and #10843

Yeah, assignment also has problem to identify the real type in such expression. Needs to be fixed as well.

**Comment by felipensp** [comment]
> Maybe can, also, fix: #16489 and #10843

It already fixes #16489 👍🏻

**Review by spytheman — APPROVED** [review]
Good work.

**Pull request merged**